integrations/slack/src/index.ts

```diff
@@ -26,8 +27,117 @@ const handleSpaceContentUpdated: EventCallback<
         return;
     }
 
+    const { data: semanticChanges } = await api.spaces.getRevisionSemanticChanges(
+        event.spaceId,
+        event.revisionId,
+        {
+            // Ignore git metadata and custom field changes
+            metadata: false,
+        }
+    );
+
+    if (semanticChanges.changes.length === 0) {
+        // No changes to notify about
+        return;
+    }
+
     const { data: space } = await api.spaces.getSpaceById(event.spaceId);
 
+    /*
+     * Build a notification that looks something like this:
+     *
+     *    Content of *Space* has been updated.
+     *
+     *    Summary of changes:
+     *    • New pages: Page1, Page2
+     *    • Modified pages: Page3
+     *    • Deleted pages: Page4, Page5
+     *    • Moved pages: Page6
+     *    • New files: File1, File2
+     *    • Modified files: File3
+     *    • Deleted files: File4
+     *
+     *    And another X changes not listed here.
+     */
+
+    const createdPages = [];
+    const editedPages = [];
+    const deletedPages = [];
+    const movedPages = [];
+    const createdFiles = [];
+    const editedFiles = [];
+    const deletedFiles = [];
+
+    semanticChanges.changes.forEach((change) => {
+        switch (change.type) {
+            case RevisionSemanticChangeType.PageCreated:
+                createdPages.push(change.page.title);
+                break;
+            case RevisionSemanticChangeType.PageEdited:
+                editedPages.push(change.page.title);
+                break;
+            case RevisionSemanticChangeType.PageDeleted:
+                deletedPages.push(change.page.title);
+                break;
+            case RevisionSemanticChangeType.PageMoved:
+                movedPages.push(change.page.title);
+                break;
+            case RevisionSemanticChangeType.FileCreated:
+                createdFiles.push(change.file.name);
+                break;
+            case RevisionSemanticChangeType.FileEdited:
+                editedFiles.push(change.file.name);
+                break;
+            case RevisionSemanticChangeType.FileDeleted:
+                deletedFiles.push(change.file.name);
+                break;
+            default:
+                break;
+        }
+    });
+
+    let notificationText = `Content of *<${space.urls.app}|${
+        space.title || 'Space'
+    }>* has been updated.`;
```

Comment:
A note here. Samy pointed out that linking to spaces may be a issue for some recipients as they may actually not have access to them. I left it as-is for now, but we could just print the space title here, without linking it.

Reply:
I'm wondering if there is any downside to them not having access when they click through. Won't it indicate that they don't have access (prompting them to either ask and admin or informing them that they can't actually look at it)? Feels odd that there is no link there when I see it in action.

Reply:
I'm gonna file a Linear issue for this topic. I don't think it should hold up pushing it out.

Reply:
I think it's a hard one to solve. We can't determine here who is going to receive the notification. I think from a UX perspective what you suggest is what makes the most sense. Accepting that some users may receive the notification and get denied and have some nice UI for it.